File: python3/Sort and Search/kth_largest_integer_min_heap.py
import heapq
from typing import List


def kth_largest_integer_min_heap(nums: List[int], k: int) -> int:
    min_heap = []
    heapq.heapify(min_heap)
    for num in nums:
        # Ensure the heap has at least 'k' integers.
        if len(min_heap) < k:
            heapq.heappush(min_heap, num)
        # If 'num' is greater than the smallest integer in the heap, pop
        # off this smallest integer from the heap and push in 'num'.
        elif num > min_heap[0]:
            heapq.heappop(min_heap)
            heapq.heappush(min_heap, num)
    return min_heap[0]

File: cpp/Sort and Search/kth_largest_integer_min_heap.cpp
#include <vector>
#include <queue>

int kthLargestIntegerMinHeap(std::vector<int>& nums, int k) {
    std::priority_queue<int, std::vector<int>, std::greater<int>> minHeap;
    for (int num : nums) {
        // Ensure the heap has at least 'k' integers.
        if (minHeap.size() < k) {
            minHeap.push(num);
        }
        // If 'num' is greater than the smallest integer in the heap, pop
        // off this smallest integer from the heap and push in 'num'.
        else if (num > minHeap.top()) {
            minHeap.pop();
            minHeap.push(num);
        }
    }
    return minHeap.top();
}